### DISCUSSION TITLE
Help! Add Brazil Device Tuya Zigbee Siren Alarm - External Converter

### DISCUSSION THREAD

**Discussion by fernandodelucca** [Koenkk/zigbee2mqtt#18649] [Devices]
[https://store.ekaza.com.br/ekaza-sirene-alarme-zigbee---ekhn-t3017/p](url)

This device is very similar to NEO Siren, but this is less tuya points available.
i need help to adjusted data types to get and set parameters.

my external converter is:

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
    // Since a lot of TuYa devices use the same modelID, but use different datapoints
    // it's necessary to provide a fingerprint instead of a zigbeeModel
    fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0601' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS0601',
            // The manufacturer name from: Device with modelID 'TS0601' is not supported.
            manufacturerName: '_TZE204_q76rtoa9',
        },
    ],
    model: 'EKHN-T3017',
    vendor: 'EKASA',
    description: 'Siren and alarm',
    fromZigbee: [tuya.fz.datapoints],
    toZigbee: [tuya.tz.datapoints],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: tuya.configureMagicPacket,
    exposes: [
        e.binary('alarm', ea.STATE_SET, true, false),
        e.enum('melody', ea.STATE_SET, Array.from(Array(18).keys()).map((x)=>(x+1).toString())),
        e.numeric('duration', ea.STATE_SET).withUnit('s').withValueMin(0).withValueMax(1800),
        e.enum('volume', ea.STATE_SET, ['low', 'medium', 'high']),
        e.enum('power_type', ea.STATE, ['battery_full', 'battery_high', 'battery_medium', 'battery_low', 'usb']),
    ],
    meta: {
        // All datapoints go in here
        tuyaDatapoints: [
            [5, 'volume', tuya.valueConverter.enum],
            [7, 'duration', tuya.valueConverter.number],
            [13, 'alarm', tuya.valueConverter.bool],
            [15, 'power_type', tuya.valueConverter.enum],
            [21, 'melody', tuya.valueConverter.enum],
        ],
    },
};

module.exports = definition;`

![image](https://github.com/Koenkk/zigbee2mqtt/assets/5623038/5db387ac-77e5-4cdd-b084-b8eabdafb6c4)

![image](https://github.com/Koenkk/zigbee2mqtt/assets/5623038/37cf71ba-d699-4cad-82c5-4340333aab25)

**Comment by felpsnew** [comment]
I have the same problem, could you help me pair.
I tried your external converter but it didn't work.
Please help me?

**Comment by felpsnew** [comment]
It is the same siren as the NEO NAS-AB02B2.
It is compatible, but the model recognized in the pairing is the TS0601.

Reference model: NAS-AB02B2 https://www.zigbee2mqtt.io/devices/NAS-AB02B2.html

EKASA model (Brazilian with neo plate): EKHN-T3017 https://store.ekaza.com.br/ekaza-sirene-alarme-zigbee---ekhn-t3017/p

If anyone has the NEO endpoints or converters, that would be great.

**Comment by felpsnew** [comment]
External converter added and normalized operation:

```javascript
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const legacy = require('zigbee-herdsman-converters/lib/legacy')
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_q76rtoa9']),
        zigbeeModel: ['q76rtoa9'],
        model: 'NAS-AB02B2',
        vendor: 'Neo',
        description: 'Alarm',
        fromZigbee: [legacy.fz.neo_alarm, fz.ignore_basic_report, fz.ignore_tuya_set_time],
        toZigbee: [legacy.tz.neo_alarm],
        exposes: [
                e.binary('alarm', ea.STATE_SET, true, false).withDescription('Enable alarm'),
                e.enum('melody', ea.STATE_SET, Array.from(Array(18).keys()).map((x)=>(x+1).toString())).withDescription('Alarm sound effect'),
                e.numeric('duration', ea.STATE_SET).withUnit('s').withValueMin(0).withValueMax(1800).withDescription('Duration time'),
                e.enum('volume', ea.STATE_SET, ['low', 'medium', 'high']).withDescription('Volume level'),
                e.numeric('battery', ea.STATE).withUnit('%').withDescription('Remaining battery in % (can take up to 24 hours before reported)'),
        ],
        onEvent: tuya.onEventSetLocalTime,
        configure: async (device, coordinatorEndpoint, logger) => {
                const endpoint = device.getEndpoint(1);
                await endpoint.command('manuSpecificTuya', 'dataQuery', {});
                await endpoint.command('manuSpecificTuya', 'mcuVersionRequest', {'seq': 0x0002});
        },
};
module.exports = definition;
```

It would be interesting to add the model and manufacturer to the NEO list.

**Reply by Koenkk** [reply]
Could you make a pull request to add out of the box support for this device?

**Comment by pabloaugusto** [comment]
Hello, I've tried @felpsnew convert file but it didn't work, many compilation errors on starting zigbee2mqtt.
So, I wrote this one based on https://github.com/Koenkk/zigbee2mqtt/issues/15718#issuecomment-1364563993, and it worked as a charm just adding {modelID: 'TS0601', manufacturerName: '_TZE204_q76rtoa9'} to list of supported devices.

@Koenkk I've opened a PR to add this device https://github.com/Koenkk/zigbee-herdsman-converters/pull/7051. Here in Basil EKAZA manufacturer is selling this model of siren with this specs.

Cheers

```javascript
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const fz = {...require('zigbee-herdsman-converters/converters/fromZigbee'), legacy: require('zigbee-herdsman-converters/lib/legacy').fromZigbee};
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_t1blo2bj'}, {modelID: 'TS0601', manufacturerName: '_TZE204_t1blo2bj'},{modelID: 'TS0601', manufacturerName: '_TZE204_q76rtoa9'}],
        zigbeeModel: ['1blo2bj'],
        model: 'NAS-AB02B2 External Test',
        vendor: 'Neo External Test',
        description: 'Alarm',
        fromZigbee: [fz.neo_alarm, fz.ignore_basic_report],
        toZigbee: [tz.neo_alarm],
        exposes: [
            e.battery_low(),
            exposes.binary('alarm', ea.STATE_SET, true, false),
            exposes.enum('melody', ea.STATE_SET, Array.from(Array(18).keys()).map((x)=>(x+1).toString())),
            exposes.numeric('duration', ea.STATE_SET).withUnit('second').withValueMin(0).withValueMax(1800),
            exposes.enum('volume', ea.STATE_SET, ['low', 'medium', 'high']),
            exposes.numeric('battpercentage', ea.STATE).withUnit('%'),
        ],
        onEvent: tuya.onEventSetLocalTime,
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await endpoint.command('manuSpecificTuya', 'dataQuery', {});
            await endpoint.command('manuSpecificTuya', 'mcuVersionRequest', {'seq': 0x0002});
        },
    };

module.exports = definition;
```

@fernandodelucca if your device still not working, try this external converter I've shared

**Reply by Danlawliet** [reply]
Tentei seguir teu comentário porem sem sucesso. Poderia me ajudar por favor, já cavei na internet toda, talvez esteja fazendo algo errado.

**Reply by pabloaugusto** [reply]
Só vi agora. Resolveu?
Qualquer coisa manda privado

**Reply by Danlawliet** [reply]
Consegui resolver sim, desculpe a demora. O Felipe me ajudou, precisava chamar o arquivo no final da config do z2m e estava com um errinho de indentação rs

**Comment by felpsnew** [comment]
fala manito, qual sirene você comprou ?

te ajudo sim, me chama ai no whatsapp da assinatura!


Abraços,

Felipe Alves

  ***@***.***>***@***.***

*Olá, me adicione no *WhatsApp +55 (61) 98465-1288
<https://wa.me/qr/TMONUWMYSRA6F1>


Em seg., 19 de fev. de 2024 às 00:24, Danlawliet ***@***.***>
escreveu:

**Reply by hbmedeiros** [reply]
Hey Felipe,

Just followed your recommendation shared by Danilo in HA Brazil and got my siren partially working, however I purchased the version with temperature and humidity sensors but those were not exposed to Z2M by default, do you know how to add them?
